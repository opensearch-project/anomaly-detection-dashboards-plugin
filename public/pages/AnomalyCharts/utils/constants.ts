/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

export const INITIAL_ANOMALY_SUMMARY = {
  anomalyOccurrence: 0,
  minAnomalyGrade: 0.0,
  maxAnomalyGrade: 0.0,
  minConfidence: 0.0,
  maxConfidence: 0.0,
  lastAnomalyOccurrence: '',
};

export enum CHART_FIELDS {
  PLOT_TIME = 'plotTime',
  ANOMALY_GRADE = 'anomalyGrade',
  CONFIDENCE = 'confidence',
  DATA = 'data',
  AGG_INTERVAL = 'aggInterval',
  EXPECTED_VALUE = 'expectedValue',
}

export enum CHART_COLORS {
  ANOMALY_GRADE_COLOR = '#D13212',
  FEATURE_DATA_COLOR = '#16191F',
  FEATURE_COLOR = '#fcd529',
  CONFIDENCE_COLOR = '#017F75',
  LIGHT_BACKGROUND = '#FFFFFF',
  DARK_BACKGROUND = '#1D1E24',
}

export const FEATURE_CHART_THEME = [
  {
    lineSeriesStyle: {
      line: {
        strokeWidth: 2,
        visible: true,
        opacity: 0.5,
      },
      point: {
        visible: true,
        stroke: CHART_COLORS.FEATURE_DATA_COLOR,
      },
    },
  },
  {
    colors: {
      vizColors: [CHART_COLORS.FEATURE_DATA_COLOR],
    },
  },
];

export const ANOMALY_CHART_THEME = [
  {
    colors: {
      vizColors: [
        CHART_COLORS.CONFIDENCE_COLOR,
        CHART_COLORS.ANOMALY_GRADE_COLOR,
      ],
    },
  },
];

export const DATE_PICKER_QUICK_OPTIONS = [
  { start: 'now-24h', end: 'now', label: 'last 24 hours' },
  { start: 'now-7d', end: 'now', label: 'last 7 days' },
  { start: 'now-30d', end: 'now', label: 'last 30 days' },
  { start: 'now-90d', end: 'now', label: 'last 90 days' },

  { start: 'now/d', end: 'now', label: 'Today' },
  { start: 'now/w', end: 'now', label: 'Week to date' },
  { start: 'now/M', end: 'now', label: 'Month to date' },
  { start: 'now/y', end: 'now', label: 'Year to date' },
];

export enum LIVE_CHART_CONFIG {
  REFRESH_INTERVAL_IN_SECONDS = 30 * 1000, //poll anomaly result every 30 seconds
  MONITORING_INTERVALS = 60,
}

export const DEFAULT_ANOMALY_SUMMARY = {
  anomalyOccurrence: 0,
  minAnomalyGrade: 0,
  maxAnomalyGrade: 0,
  minConfidence: 0,
  maxConfidence: 0,
  lastAnomalyOccurrence: '-',
};

export const HEATMAP_CHART_Y_AXIS_WIDTH = 30;
