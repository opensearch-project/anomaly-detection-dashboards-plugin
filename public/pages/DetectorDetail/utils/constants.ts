/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

export enum DETECTOR_DETAIL_TABS {
  RESULTS = 'results',
  HISTORICAL = 'historical',
  CONFIGURATIONS = 'configurations',
}

const DEFAULT_ACTION_ITEM = 'Restart the detector and try again.';
// Known causes:
// https://github.com/opensearch-project/anomaly-detection/blob/development/src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyResultTransportAction.java#L174-L185
export const DETECTOR_INIT_FAILURES = Object.freeze({
  NO_TRAINING_DATA: {
    //https://github.com/opensearch-project/anomaly-detection/blob/development/src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyResultTransportAction.java#L801
    keyword: 'Cannot get training data',
    cause: 'no sufficient data is ingested',
    actionItem:
      'Make sure your data is ingested correctly. If your data source has infrequent ingestion, increase the detector time interval and try again.',
  },
  COLD_START_ERROR: {
    //https://github.com/opensearch-project/anomaly-detection/blob/development/src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyResultTransportAction.java#L811
    keyword: 'Error while cold start',
    cause: 'of an error during model training',
    actionItem: DEFAULT_ACTION_ITEM,
  },
  AD_MODEL_MEMORY_REACH_LIMIT: {
    //https://github.com/opensearch-project/anomaly-detection/blob/development/src/main/java/com/amazon/opendistroforelasticsearch/ad/ml/ModelManager.java#L272
    keyword: 'AD models memory usage exceeds our limit',
    cause: 'of lack of memory for the detector models',
    actionItem: 'Reduce the number of features and try again.',
  },
  DETECTOR_MEMORY_REACH_LIMIT: {
    //https://github.com/opensearch-project/anomaly-detection/blob/development/src/main/java/com/amazon/opendistroforelasticsearch/ad/ml/ModelManager.java#L783
    keyword: 'Exceeded memory limit',
    cause: 'of lack of memory for the detector',
    actionItem:
      "Remove or stop other detectors that you don't actively use, increase your cluster size, reduce the number of features, or scale up with an instance type of more memory and try again.",
  },
  DATA_INDEX_NOT_FOUND: {
    //https://github.com/opensearch-project/anomaly-detection/blob/development/src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyResultTransportAction.java#L366
    keyword: 'Having trouble querying data: ',
    cause: 'the data index is not found',
    actionItem: 'Make sure your index exists and try again.',
  },
  ALL_FEATURES_DISABLED: {
    //https://github.com/opensearch-project/anomaly-detection/blob/development/src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyResultTransportAction.java#L368
    keyword:
      'Having trouble querying data because all of your features have been disabled',
    cause: 'all detector features are disabled',
    actionItem: 'Enable one or more features and try again.',
  },
  DETECTOR_UNDEFINED: {
    //https://github.com/opensearch-project/anomaly-detection/blob/development/src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyResultTransportAction.java#L230
    keyword: 'AnomalyDetector is not available',
    cause: 'the detector is not defined',
    actionItem: 'Define your detector and try again.',
  },
  INVALID_FEATURE_QUERY: {
    // TODO: https://github.com/opensearch-project/anomaly-detection/issues/218
    //https://github.com/opensearch-project/anomaly-detection/blob/master/src/main/java/com/amazon/opendistroforelasticsearch/ad/constant/CommonErrorMessages.java#L28
    keyword: 'Invalid search query',
    cause: 'the custom query caused an invalid feature aggregation',
    actionItem:
      'Revise the custom query in your feature aggregations and try again.',
  },
  UNKNOWN_EXCEPTION: {
    //https://github.com/opensearch-project/anomaly-detection/blob/development/src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/AnomalyResultTransportAction.java#L438
    keyword: 'We might have bugs',
    cause: 'of unknown error',
    actionItem: DEFAULT_ACTION_ITEM,
  },
});
