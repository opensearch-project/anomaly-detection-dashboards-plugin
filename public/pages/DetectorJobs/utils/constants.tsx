/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import { DetectorJobsFormikValues } from '../../DetectorJobs/models/interfaces';

export const INITIAL_DETECTOR_JOB_VALUES: DetectorJobsFormikValues = {
  realTime: true,
  historical: false,
  startTime: 'now-30d',
  endTime: 'now',
};

export const HISTORICAL_DATE_RANGE_COMMON_OPTIONS = [
  { start: 'now-24h', end: 'now', label: 'last 24 hours' },
  { start: 'now-7d', end: 'now', label: 'last 7 days' },
  { start: 'now-30d', end: 'now', label: 'last 30 days' },
  { start: 'now-90d', end: 'now', label: 'last 90 days' },

  { start: 'now/d', end: 'now', label: 'Today' },
  { start: 'now/w', end: 'now', label: 'Week to date' },
  { start: 'now/M', end: 'now', label: 'Month to date' },
  { start: 'now/y', end: 'now', label: 'Year to date' },
];
