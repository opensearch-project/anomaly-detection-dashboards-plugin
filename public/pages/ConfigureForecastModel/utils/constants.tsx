/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import {
  ModelConfigurationFormikValues,
} from '../../ConfigureForecastModel/models/interfaces';
import { DEFAULT_SHINGLE_SIZE } from '../../../utils/constants';

// FIXME: We intentionally leave some parameters undefined to encourage customers to click 
// the "Suggest parameters" button rather than providing default values like "10 minutes" for interval
export const INITIAL_MODEL_CONFIGURATION_VALUES: ModelConfigurationFormikValues =
  {
    shingleSize: DEFAULT_SHINGLE_SIZE,
    imputationOption: undefined,
    interval: undefined,
    windowDelay: undefined,
    suggestedSeasonality: undefined,
    recencyEmphasis: undefined,
    resultIndex: undefined,
    resultIndexMinAge: 7,
    resultIndexMinSize: 51200,
    resultIndexTtl: 60,
    flattenCustomResultIndex: false,
  };

// an enum for the sparse data handling options
export enum SparseDataOptionValue {
  IGNORE = 'ignore',
  PREVIOUS_VALUE = 'previous_value',
  SET_TO_ZERO = 'set_to_zero',
  CUSTOM_VALUE = 'custom_value',
}
