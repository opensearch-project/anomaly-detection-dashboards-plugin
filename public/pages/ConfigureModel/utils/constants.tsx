/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import { FEATURE_TYPE } from '../../../models/interfaces';
import {
  ModelConfigurationFormikValues,
  FeaturesFormikValues,
} from '../../ConfigureModel/models/interfaces';
import { DEFAULT_SHINGLE_SIZE } from '../../../utils/constants';

export const INITIAL_MODEL_CONFIGURATION_VALUES: ModelConfigurationFormikValues =
  {
    featureList: [],
    categoryFieldEnabled: false,
    categoryField: [],
    shingleSize: DEFAULT_SHINGLE_SIZE,
    imputationOption: undefined
  };

export const INITIAL_FEATURE_VALUES: FeaturesFormikValues = {
  featureId: '',
  featureName: '',
  featureEnabled: true,
  featureType: FEATURE_TYPE.SIMPLE,
  aggregationQuery: JSON.stringify(
    { aggregation_name: { sum: { field: 'field_name' } } },
    null,
    4
  ),
  aggregationBy: '',
  aggregationOf: [],
};

export const FEATURE_TYPES = [
  { text: 'Custom Aggregation', value: FEATURE_TYPE.CUSTOM },
  { text: 'Defined Aggregation', value: FEATURE_TYPE.SIMPLE },
];

export const FEATURE_TYPE_OPTIONS = [
  { text: 'Field value', value: FEATURE_TYPE.SIMPLE },
  { text: 'Custom expression', value: FEATURE_TYPE.CUSTOM },
];

export enum SAVE_FEATURE_OPTIONS {
  START_AD_JOB = 'start_ad_job',
  KEEP_AD_JOB_STOPPED = 'keep_ad_job_stopped',
}

export const AGGREGATION_TYPES = [
  { value: 'avg', text: 'average()' },
  { value: 'value_count', text: 'count()' },
  { value: 'sum', text: 'sum()' },
  { value: 'min', text: 'min()' },
  { value: 'max', text: 'max()' },
];

export const FEATURE_FIELDS = [
  'featureName',
  'aggregationOf',
  'aggregationBy',
  'aggregationQuery',
];

// an enum for the sparse data handling options
export enum SparseDataOptionValue {
  IGNORE = 'ignore',
  PREVIOUS_VALUE = 'previous_value',
  SET_TO_ZERO = 'set_to_zero',
  CUSTOM_VALUE = 'custom_value',
}
