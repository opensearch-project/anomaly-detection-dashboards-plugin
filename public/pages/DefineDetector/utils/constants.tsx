/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import { FILTER_TYPES, UIFilter } from '../../../models/interfaces';
import { OPERATORS_MAP } from '../../DefineDetector/components/DataFilterList/utils/constant';
import { DetectorDefinitionFormikValues } from '../../DefineDetector/models/interfaces';
export const EMPTY_UI_FILTER: UIFilter = {
  filterType: FILTER_TYPES.SIMPLE,
  fieldInfo: [],
  operator: OPERATORS_MAP.IS,
  fieldValue: '',
  query: '',
  label: '',
};

export const testDetectorDefinitionValues: DetectorDefinitionFormikValues = {
  name: 'test',
  description: 'desc',
  index: [],
  filters: [],
  filterQuery: JSON.stringify({ bool: { filter: [] } }, null, 4),
  timeField: 'timestamp',
  interval: 10,
  windowDelay: 1,
  resultIndex: undefined,
};

export const INITIAL_DETECTOR_DEFINITION_VALUES: DetectorDefinitionFormikValues =
  {
    name: '',
    description: '',
    index: [],
    filters: [],
    filterQuery: JSON.stringify({ bool: { filter: [] } }, null, 4),
    timeField: '',
    interval: 10,
    windowDelay: 1,
    resultIndex: undefined,
    resultIndexMinAge: 7,
    resultIndexMinSize: 51200,
    resultIndexTtl: 60,
    flattenCustomResultIndex: false,
    imputationMethod: undefined,
    customImputationValue: undefined
  };
