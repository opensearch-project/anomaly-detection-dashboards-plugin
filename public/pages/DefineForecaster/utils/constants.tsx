/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import { FILTER_TYPES, UIFilter } from '../../../models/interfaces';
import { OPERATORS_MAP } from '../../DefineDetector/components/DataFilterList/utils/constant';
import { ForecasterDefinitionFormikValues } from '../models/interfaces';
import { FEATURE_TYPE } from '../../../models/interfaces';
import {
  FeaturesFormikValues,
} from '../../ConfigureModel/models/interfaces';

export const EMPTY_UI_FILTER: UIFilter = {
  filterType: FILTER_TYPES.SIMPLE,
  fieldInfo: [],
  operator: OPERATORS_MAP.IS,
  fieldValue: '',
  query: '',
  label: '',
};

export const testForecasterDefinitionValues: ForecasterDefinitionFormikValues = {
  name: 'test',
  description: 'desc',
  index: [],
  filters: [],
  filterQuery: JSON.stringify({ bool: { filter: [] } }, null, 4),
  timeField: 'timestamp',
  featureList: [],
  categoryFieldEnabled: false,
  categoryField: [],
};

export const INITIAL_FORECASTER_DEFINITION_VALUES: ForecasterDefinitionFormikValues =
  {
    name: '',
    description: '',
    index: [],
    filters: [],
    filterQuery: JSON.stringify({ bool: { filter: [] } }, null, 4),
    timeField: '',
    featureList: [],
    categoryFieldEnabled: false,
    categoryField: [],
  };

  export const FEATURE_TYPES = [
    { text: 'Custom Aggregation', value: FEATURE_TYPE.CUSTOM },
    { text: 'Defined Aggregation', value: FEATURE_TYPE.SIMPLE },
  ];

  export const FEATURE_TYPE_OPTIONS = [
    { text: 'Field value', value: FEATURE_TYPE.SIMPLE },
    { text: 'Custom expression', value: FEATURE_TYPE.CUSTOM },
  ];

  export const INITIAL_FEATURE_VALUES: FeaturesFormikValues = {
    featureId: '',
    featureName: '',
    featureEnabled: true,
    featureType: FEATURE_TYPE.SIMPLE,
    aggregationQuery: JSON.stringify(
      { aggregation_name: { sum: { field: 'field_name' } } },
      null,
      4
    ),
    aggregationBy: '',
    aggregationOf: [],
  };

  export const AGGREGATION_TYPES = [
    { value: 'avg', text: 'average()' },
    { value: 'value_count', text: 'count()' },
    { value: 'sum', text: 'sum()' },
    { value: 'min', text: 'min()' },
    { value: 'max', text: 'max()' },
  ];

