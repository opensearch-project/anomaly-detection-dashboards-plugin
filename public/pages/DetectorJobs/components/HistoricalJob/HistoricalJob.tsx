/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import {
  EuiText,
  EuiLink,
  EuiIcon,
  EuiFlexItem,
  EuiFlexGroup,
  EuiCheckbox,
  EuiSuperDatePicker,
} from '@elastic/eui';
import { Field, FieldProps, FormikProps } from 'formik';
import { get } from 'lodash';
import React, { useState } from 'react';
import ContentPanel from '../../../../components/ContentPanel/ContentPanel';
import { FormattedFormRow } from '../../../../components/FormattedFormRow/FormattedFormRow';
import { DetectorJobsFormikValues } from '../../models/interfaces';
import { HISTORICAL_DATE_RANGE_COMMON_OPTIONS } from '../../utils/constants';
import { BASE_DOCS_LINK } from '../../../../utils/constants';
import {
  isInvalid,
  getError,
  convertTimestampToString,
} from '../../../../utils/utils';

interface HistoricalJobProps {
  formikProps: FormikProps<DetectorJobsFormikValues>;
  setHistorical(historical: boolean): void;
}
export function HistoricalJob(props: HistoricalJobProps) {
  const [enabled, setEnabled] = useState<boolean>(
    get(props, 'formikProps.values.historical', true)
  );

  return (
    <ContentPanel
      title="Historical analysis detection"
      titleSize="s"
      subTitle={
        <EuiText
          className="content-panel-subTitle"
          style={{ lineHeight: 'normal' }}
        >
          Historical analysis detection lets you analyze and apply machine
          learning models over long historical data windows (weeks or months).
          You can identify anomaly patterns, seasonality, and trends.{' '}
          <EuiLink href={`${BASE_DOCS_LINK}/ad`} target="_blank">
            Learn more <EuiIcon size="s" type="popout" />
          </EuiLink>
        </EuiText>
      }
    >
      <Field name="historical">
        {({ field, form }: FieldProps) => (
          <EuiFlexGroup direction="column">
            <EuiFlexItem>
              <EuiCheckbox
                id={'historicalCheckbox'}
                label="Run historical analysis detection"
                checked={enabled}
                onChange={() => {
                  if (!enabled) {
                    props.setHistorical(true);
                  }
                  if (enabled) {
                    props.setHistorical(false);
                  }
                  setEnabled(!enabled);
                }}
              />
            </EuiFlexItem>
            {enabled ? (
              <EuiFlexItem>
                <FormattedFormRow
                  title="Historical analysis date range"
                  helpText="Select a date range for your historical analysis (you may adjust later)."
                  isInvalid={isInvalid(field.name, form)}
                  error={getError(field.name, form)}
                >
                  <EuiSuperDatePicker
                    //isLoading={props.isLoading}
                    start={convertTimestampToString(form.values.startTime)}
                    end={convertTimestampToString(form.values.endTime)}
                    onTimeChange={({
                      start,
                      end,
                      isInvalid,
                      isQuickSelection,
                    }) => {
                      form.setFieldValue('startTime', start);
                      form.setFieldValue('endTime', end);
                    }}
                    isPaused={true}
                    showUpdateButton={false}
                    commonlyUsedRanges={HISTORICAL_DATE_RANGE_COMMON_OPTIONS}
                  />
                </FormattedFormRow>
              </EuiFlexItem>
            ) : null}
          </EuiFlexGroup>
        )}
      </Field>
    </ContentPanel>
  );
}
