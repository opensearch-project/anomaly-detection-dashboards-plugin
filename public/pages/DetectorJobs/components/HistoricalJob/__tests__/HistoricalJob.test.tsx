/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from 'react';
import { render } from '@testing-library/react';
import { HistoricalJob } from '../HistoricalJob';

import { Formik } from 'formik';
import { INITIAL_DETECTOR_JOB_VALUES } from '../../../utils/constants';

describe('<HistoricalJob /> spec', () => {
  test('renders the component', () => {
    const { container } = render(
      <Formik initialValues={{ detectorName: '' }} onSubmit={jest.fn()}>
        {() => (
          <div>
            <HistoricalJob
              formikProps={{
                values: { ...INITIAL_DETECTOR_JOB_VALUES, historical: false },
              }}
            />
          </div>
        )}
      </Formik>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  test('hides date picker if disabled', async () => {
    const { queryByText } = render(
      <Formik initialValues={{ detectorName: '' }} onSubmit={jest.fn()}>
        {() => (
          <div>
            <HistoricalJob
              formikProps={{
                values: { ...INITIAL_DETECTOR_JOB_VALUES, historical: false },
              }}
            />
          </div>
        )}
      </Formik>
    );
    expect(queryByText('Run historical analysis detection')).not.toBeNull();
    expect(queryByText('Historical analysis date range')).toBeNull();
  });
  test('shows date picker if enabled', async () => {
    const { queryByText } = render(
      <Formik initialValues={{ detectorName: '' }} onSubmit={jest.fn()}>
        {() => (
          <div>
            <HistoricalJob
              formikProps={{
                values: { ...INITIAL_DETECTOR_JOB_VALUES, historical: true },
              }}
            />
          </div>
        )}
      </Formik>
    );
    expect(queryByText('Run historical analysis detection')).not.toBeNull();
    expect(queryByText('Historical analysis date range')).not.toBeNull();
  });
});
