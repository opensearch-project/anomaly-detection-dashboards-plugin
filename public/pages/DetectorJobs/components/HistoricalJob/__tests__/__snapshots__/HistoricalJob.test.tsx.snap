// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HistoricalJob /> spec renders the component 1`] = `
<div>
  <div
    class="euiPanel euiPanel--paddingMedium"
    style="padding: 20px;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 0px;"
    >
      <div
        class="euiFlexItem"
      >
        <h3
          class="euiTitle euiTitle--small"
        >
          Historical analysis detection
        </h3>
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem content-panel-subTitle"
            style="line-height: normal; max-width: 75%;"
          >
            <div
              class="euiText euiText--medium content-panel-subTitle"
              style="line-height: normal;"
            >
              Historical analysis detection lets you analyze and apply machine learning models over long historical data windows (weeks or months). You can identify anomaly patterns, seasonality, and trends.
               
              <a
                class="euiLink euiLink--primary"
                href="https://opensearch.org/docs/monitoring-plugins/ad"
                rel="noopener noreferrer"
                target="_blank"
              >
                Learn more 
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--small euiIcon-isLoading"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem euiFlexItem--flexGrowZero"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
          />
        </div>
      </div>
    </div>
    <div>
      <hr
        class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginSmall"
      />
      <div
        style="padding: 10px 0px;"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionColumn euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
          >
            <div
              class="euiCheckbox"
            >
              <input
                class="euiCheckbox__input"
                id="historicalCheckbox"
                type="checkbox"
              />
              <div
                class="euiCheckbox__square"
              />
              <label
                class="euiCheckbox__label"
                for="historicalCheckbox"
              >
                Run historical analysis detection
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
